### DISCUSSION TITLE
support workspaces in audit

### DISCUSSION THREAD

**Discussion by MarkSheinkman** [npm/feedback#333] [CLI] [answered]
I'm not seeing any discussions or issues about this, not any explicit indication in the docs about it not being planned, so not sure if it's a bug or a feature request. 

Let's say I have 2 workspaces:
```
package.json ("workspaces": ["a", "b"])
a\package.json
b\package.json ("dependencies": {"vulnerable-package": "1"})
```

I'd like to audit all the packages referenced in the repo. Ideally, `npm audit` in the repo root should do just that, but it's not reporting `vulnerable-package` because it's not in the root `package.json`.
If I audit in on of the workspaces, it fails for not having a `package-lock.json` file. I don't want to generate one on the spot since it could resolve different dependencies that the root one - e.g when `a` needs `1||2||4`, `b` needs `2||3`, and version 2 has a vulnerability. 

Currently I work around that by copying `package-lock.json` from the root to each workspace and only then running an audit, but that's cumbersome and hacky. Feels like there should be a better solution.

**Comment by MylesBorins** [comment, marked as answer]
I know the @npm/cli team has been doing quite a bit on improving workspaces including making various commands workspace aware

[This RFC](https://github.com/npm/rfcs/pull/117) outlines various commands to be eventually supported including Audit.

**Reply by MarkSheinkman** [reply]
Thanks! glad to hear it's in progress.
